napari/settings/_appearance.py

```diff
@@ -7,7 +7,33 @@
 from napari.utils.translations import trans
 
 
+class HighlightSettings(EventedModel):
+    highlight_thickness: int = Field(
+        1,
+        title=trans._('Highlight thickness'),
+        description=trans._(
+            'Select the highlight thickness when hovering over shapes/points.'
+        ),
+        ge=1,
+        le=10,
+    )
+    highlight_color: List[float] = Field(
+        [0.0, 0.6, 1.0, 1.0],
+        title=trans._('Highlight color'),
+        description=trans._(
+            'Select the highlight color when hovering over shapes/points.'
+        ),
+    )
+
+
 class AppearanceSettings(EventedModel):
+    def __init__(self, **kwargs: Any) -> None:
+        super().__init__(**kwargs)
+        self.highlight.events.connect(self._highlight_changed)
+
+    def _highlight_changed(self) -> None:
+        self.events.highlight(value=self.highlight)
```

Comment:
Any worries about this internal connection @Czaki ?

Reply:
I do not understand why it is needed. 
The only possibility is that we do something wrong in backend and there is lack of proper event emission.